x-pack/legacy/plugins/siem/public/components/links/index.tsx

```diff
@@ -117,34 +125,102 @@ export const CertificateFingerprintLink = React.memo<{
-        domain
-      )}`}
-      target="_blank"
-    >
-      {children ? children : domain}
-    </EuiLink>
-  )
-);
+enum DefaultReputationLink {
+  'virustotal.com' = 'virustotal.com',
+  'talosIntelligence.com' = 'talosIntelligence.com',
+}
+
+export interface ReputationLinkSetting {
+  name: string;
+  url_template: string;
+}
+
+function isDefaultReputationLink(name: string): name is DefaultReputationLink {
+  return (
+    name === DefaultReputationLink['virustotal.com'] ||
+    name === DefaultReputationLink['talosIntelligence.com']
+  );
+}
+const isReputationLink = (
+  rowItem: string | ReputationLinkSetting
+): rowItem is ReputationLinkSetting =>
+  (rowItem as ReputationLinkSetting).url_template !== undefined &&
+  (rowItem as ReputationLinkSetting).name !== undefined;
+
+export const DEFAULT_NUMBER_OF_REPUTATION_LINK = 5;
+const ReputationLinkComponent: React.FC<{
+  overflowIndexStart?: number;
+  allItemsLimit?: number;
+  showDomain?: boolean;
+  domain: string;
+}> = ({
+  overflowIndexStart = DEFAULT_NUMBER_OF_REPUTATION_LINK,
+  allItemsLimit = DEFAULT_NUMBER_OF_REPUTATION_LINK,
+  showDomain = false,
+  domain,
+}) => {
+  const [ipReputationLinksSetting] = useUiSetting$<ReputationLinkSetting[]>(
+    IP_REPUTATION_LINKS_SETTING
+  );
+  const [ipReputationLinks, setIpReputationLinks] = useState<ReputationLinkSetting[]>(
+    ipReputationLinksSetting
+  );
+
+  const defaultNameMapping: Record<DefaultReputationLink, string> = useMemo(
+    () => ({
+      [DefaultReputationLink['virustotal.com']]: i18n.VIEW_VIRUS_TOTAL,
+      [DefaultReputationLink['talosIntelligence.com']]: i18n.VIEW_TALOS_INTELLIGENCE,
+    }),
+    []
+  );
+
+  useEffect(() => {
+    setIpReputationLinks(
+      ipReputationLinks
+        ?.slice(0, allItemsLimit)
+        .filter(({ url_template }) => !isUrlInvalid(url_template))
+        .map(({ name, url_template }: { name: string; url_template: string }) => {
+          return {
+            name: isDefaultReputationLink(name) ? defaultNameMapping[name] : name,
+            url_template: url_template.replace(`{{ip}}`, encodeURIComponent(domain)),
+          };
+        })
+    );
+  }, [domain, overflowIndexStart, setIpReputationLinks, defaultNameMapping]);
+
+  return (
+    <>
+      {ipReputationLinks
+        ?.slice(0, overflowIndexStart)
+        .map(({ name, url_template: urlTemplate }: ReputationLinkSetting, id) => (
+          <EuiLink href={urlTemplate} target="_blank" key={`reputationLink-${id}`}>
+            {showDomain ? domain : name ?? domain}
+            {id !== Math.max(0, Math.min(overflowIndexStart, ipReputationLinks.length) - 1) && ', '}
+          </EuiLink>
+        ))}
+      <DefaultFieldRendererOverflow
+        rowItems={ipReputationLinks}
+        idPrefix="moreReputationLink"
+        render={rowItem => {
+          return (
+            isReputationLink(rowItem) && (
+              <EuiLink href={rowItem.url_template} target="_blank">
```

Comment:
Consider adding `rel="noopener"` to prevent linked-to pages from accessing `window.opener`